### ISSUE TITLE
Improve spring rest parsing

### ISSUE THREAD

**Issue by H3llK33p3r** [vojtechhabarta/typescript-generator#343]
Hi, I'm trying to use the generator to generate an axios client over Spring Rest Controller. 
I notice that the request parameter names are only mapped when value is used in the annotation not the name (alias for value). 

This is due to the following code in SpringApplicationParser : 

```
for (Parameter param : method.getParameters()) {
                final RequestParam requestParamAnnotation = param.getAnnotation(RequestParam.class);
                if (requestParamAnnotation != null) {
                    queryParams.add(new RestQueryParam.Single(new MethodParameterModel(requestParamAnnotation.value(), param.getParameterizedType())));
                    foundType(result, param.getParameterizedType(), controllerClass, method.getName());
                }
            }
``` 

The "issue" should also be present for path variable. 
Can we consider to change the code to take either the value or the name because in this case the code generated has empty typescript parameter name. 

Example for 
```java
	@GetMapping(path = "api/path")
	public String helloworld(@PathVariable("applicationId") String applicationId,
			@RequestParam(name = "start", required = false, defaultValue = "0",value ="start" ) Integer start,
			@RequestParam(name = "count", required = false, defaultValue = "0") Integer count) {
		return "helloworld";
	}
```

```typescript
export class RestApplicationClient<O> {

    constructor(protected httpClient: HttpClient<O>) {
    }

    /**
     * HTTP GET /api/path
     * Java method: com.example.test.TestWS.helloworld
     */
    helloworld(queryParams?: { start?: number; ""?: number; }, options?: O): RestResponse<string> {
        return this.httpClient.request({ method: 'GET', url: uriEncoding`api/path`, queryParams: queryParams, options: options });
    }
}
```

Sorry I'm not able to do a merge request to that :( . 
Thank you in advance for the reply :) !

**Comment by vojtechhabarta** [comment]
@H3llK33p3r thanks for reporting this (with example).
I will check how exactly Spring behaves and I will fix it.

**Comment by Antibrumm** [comment]
You should be able to follow the `@AliasFor` annotation. Spring uses it to identify aliases like `name()/value()` in `@RequestParam` or `path()/value()` in `@RequestMapping`.

You might be able to use the AnnotationUtils directly so you don't need to fiddle out the aliases yourself: (from 

```
@AliasFor
  <h3>Usage Requirements</h3>
  <p>Like with any annotation in Java, the mere presence of {@code @AliasFor}
   on its own will not enforce alias semantics. For alias semantics to be
   enforced, annotations must be <em>loaded</em> via the utility methods in
   {@link AnnotationUtils}. Behind the scenes, Spring will <em>synthesize</em>
   an annotation by wrapping it in a dynamic proxy that transparently enforces
   <em>attribute alias</em> semantics for annotation attributes that are
   annotated with {@code @AliasFor}. Similarly, {@link AnnotatedElementUtils}
   supports explicit meta-annotation attribute overrides when {@code @AliasFor}
   is used within an annotation hierarchy. Typically you will not need to
   manually synthesize annotations on your own since Spring will do that for
   you transparently when looking up annotations on Spring-managed components.
```

**Comment by vojtechhabarta** [comment]
Thanks @Antibrumm I was just checking `AnnotatedElementUtils` if there is something I could use. I will also check `AnnotationUtils`.

**Comment by vojtechhabarta** [comment]
I fixed query params (and other places) by using `AnnotationUtils.findAnnotation()`.

**Comment by vojtechhabarta** [comment]
Released in [2.15](https://github.com/vojtechhabarta/typescript-generator/releases).
